### ISSUE TITLE
GetValueByPointerWithDefault() take const document?

### ISSUE THREAD

**Issue by xlqian** [miloyip/rapidjson#837]
It's curious that GetValueByPointerWithDefault() doesn't take const document meanwhile GetValueByPointer() does take const document... 

Why? 

Thx

**Comment by miloyip** [comment]
Yes. Because there is combinatorial explosion... It is not technically a problem.
